stdlib/Dates/src/periods.jl

```diff
@@ -351,6 +354,14 @@ Base.show(io::IO,x::CompoundPeriod) = print(io, string(x))
 (-)(x::CompoundPeriod) = CompoundPeriod(-x.periods)
 (-)(y::Union{Period, CompoundPeriod}, x::CompoundPeriod) = (-x) + y
 
+(*)(x::CompoundPeriod, y::Real) = iszero(y) ? CompoundPeriod() : CompoundPeriod(x.periods * y)
+(*)(y::Real, x::CompoundPeriod) = x * y
+
+(*)(A::CompoundPeriod, B::AbstractArray) = Broadcast.broadcast_preserving_zero_d(*, A, B)
+(*)(A::AbstractArray, B::CompoundPeriod) = Broadcast.broadcast_preserving_zero_d(*, A, B)
+
+(/)(x::CompoundPeriod, y::Real) = isfinite(y) ? CompoundPeriod(x.periods / y) : CompoundPeriod()
```

Comment:
Do you mean `!isinf` instead of `isfinite`?  The `isfinite` function also returns `false` for `NaN`

Reply:
And do we also need a `(\)(y::Real, x::CompoundPeriod) = x / y` method?

Reply:
I didn’t think about the `NaN` case, I will change it to `!isinf`.

`\` currently isn’t implemented for `Period`s, so I didn’t add it for `CompoundPeriod`. But I think it would be good to add both.

Reply:
I have fixed the `NaN` case. I would leave `\` for another PR, since it is currently not implemented for `Period`s.